### PR TITLE
[IOTDB-1087] Flush thread hang in the merge method of LevelCompaction…

### PR DESCRIPTION
**Pull request by yuqi1129**
…TsFileManagement

### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/engine/compaction/IoTDBCompressReferenceBugTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    EnvironmentUtils.envSetUp();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    EnvironmentUtils.cleanEnv();
+  }
+
+  @Test
+  public void testGZIPCompression() throws Exception {
```
Comment (HTHou) [inline server/src/test/java/org/apache/iotdb/db/engine/compaction/IoTDBCompressReferenceBugTest.java:52]: testGZIPCompression???

**Comment by zhanglingzhe0820** [comment]
It seems that you fix this bug by modify lock, but it is not a lock related bug. We are fixing this bug in pr:https://github.com/apache/iotdb/pull/2341 . Can you just help us review this pr as it cannot pass windows ci.

File: server/src/test/java/org/apache/iotdb/db/engine/compaction/IoTDBCompressReferenceBugTest.java (hunk 1 above)
Comment (yuqi1129) [reply server/src/test/java/org/apache/iotdb/db/engine/compaction/IoTDBCompressReferenceBugTest.java:52]: I directly move the test from [pr#2264](https://github.com/apache/iotdb/pull/2264) and forget to rename the method name, I will change it later

**Comment by yuqi1129** [comment]
@zhanglingzhe0820, OK, i will make some test about your PR, thanks for your attention

**Comment by zhanglingzhe0820** [comment]
as the bug is fixed in pr:https://github.com/apache/iotdb/pull/2341, and it is merged, shall we close this pr

**Comment by yuqi1129** [comment]
@zhanglingzhe0820 , OK, i will close it later

**Pull request closed without merging**